perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_def_googleGroup_namespace_cz.java

```diff
@@ -0,0 +1,53 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.core.api.ActionType;
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AuthzResolver;
+import cz.metacentrum.perun.core.api.Group;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeValueException;
+import cz.metacentrum.perun.core.api.exceptions.WrongReferenceAttributeValueException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleAbstract;
+import cz.metacentrum.perun.core.implApi.modules.attributes.GroupAttributesModuleImplApi;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Group googleGroup-namespace attribute.
+ *
+ * @author Michal Holič  holic.michal@gmail.com
+ */
+public class urn_perun_group_attribute_def_def_googleGroup_namespace_cz extends GroupAttributesModuleAbstract implements GroupAttributesModuleImplApi {
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl sess, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		//prepare groupName value variable
+		String groupName = null;
+		if(attribute.getValue() != null) groupName = (String) attribute.getValue();
+
+		if(groupName == null) {
+			// if this is group attribute, its ok
+			return;
+		}else if(!groupName.matches("^[-_a-zA-Z0-9]+$")){
+			throw new WrongAttributeValueException(attribute,"GroupName attributte content invalid characters. Allowed are only letters, numbers and characters _ and -.");
+		}
+
+		//TODO Check reserved google group names
+		//sess.getPerunBl().getModulesUtilsBl().checkReservedGoogleGroupNames(attribute);
+
+		//prepare lists of groups with the same groupName value in the same namespace
+		List<Group> groupsWithSameGroupNameInTheSameNamespace = new ArrayList<Group>();
+
+		//Fill lists of groups
+		groupsWithSameGroupNameInTheSameNamespace.addAll(sess.getPerunBl().getGroupsManagerBl().getGroupsByAttribute(sess, attribute));
+
+		//If there is no group with same GroupNameInTheSameNamespace, its ok
+		if(groupsWithSameGroupNameInTheSameNamespace.isEmpty()) return;
```

Comment:
Be carefull, it can possibly also find this group. Better will be use
groupsWithSameGroupNameInTheSameNamespace.remove(group) for sure.